pkg/preparer/setup.go

```diff
@@ -52,6 +52,10 @@ type PreparerConfig struct {
 	WriteKVHealth        bool                   `yaml:"write_kv_health,omitempty"`
 	UseSessionHealth     bool                   `yaml:"use_session_health,omitempty"`
 	LogLevel             string                 `yaml:"log_level,omitempty"`
+
+	// Params defines a collection of miscellaneous runtime parameters defined throughout the
+	// source files.
+	Params map[string]string `yaml:"params"`
```

Comment:
by itself I'm concerned that a user of this might be tempted to just read from `config.Params` directly instead of using the `params` package. Perhaps typedef to `params.Map` or something to make it clear that it's used in conjunction with that package?